src/ipld/cid_vec.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::utils::cid::{CidVariant, BLAKE2B256_SIZE};
+use cid::{
+    multihash::{self, Code::Blake2b256},
+    Cid,
+};
+use fvm_ipld_encoding::DAG_CBOR;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+// CidVec takes advantage of the fact that the V1 DAG-CBOR Blake2b-256 variant (which can be stored in 32 bytes vs 96 bytes for a `Cid` type) is +99.99% of all CIDs. CidVec defaults to the `Vec<[u8; BLAKE2B256_SIZE]>` type, only using the more expensive `Vec<Cid>` type when necessary.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum CidVec {
```

Comment:
I also don't think this should be an `enum` - users can see an implementation detail

Reply:
No longer applicable per https://chainsafeio.slack.com/archives/C0144LHN732/p1691054770109249?thread_ts=1690999613.526739&cid=C0144LHN732.